src/App.tsx

```diff
@@ -1,24 +1,196 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useRef, useState } from 'react';
 import { UserWarning } from './UserWarning';
+import { addTodos, deleteTodos, getTodos } from './api/todos';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { ErrorTypes } from './types/ErrorTypes';
+import { ErrorMessage } from './components/ErrorMessage';
+import { TodoItem } from './components/TodoItem';
+import { FilterTypes } from './types/FilterTypes';
 
-const USER_ID = 0;
+const USER_ID = 10905;
 
 export const App: React.FC = () => {
+  const [title, setTitle] = useState<string>('');
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedFilter, setSelectedFilter] = useState(FilterTypes.ALL);
+  const [loadingTodosId, setLoadingTodosId] = useState<number[]>([0]);
+  const [isDisabled, setIsDisabled] = useState(false);
+  const [isError, setIsError] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [errorMessage, setErrorMessage] = useState<ErrorTypes>();
+
+  const titleField = useRef<HTMLInputElement>(null);
+
+  const handleChangeTitle = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(e.target.value);
+  };
+
+  const showAndDeleteError = () => {
+    setIsError(true);
+
+    return setTimeout(() => setIsError(false), 3000);
+  };
+
+  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    if (!title.trim()) {
+      setErrorMessage(ErrorTypes.TITLE);
+      setTitle('');
+
+      return showAndDeleteError();
+    }
+
+    setIsDisabled(true);
+    setTempTodo({
+      id: 0,
+      userId: USER_ID,
+      title,
+      completed: false,
+    });
+
+    addTodos({
+      title,
+      userId: USER_ID,
+      completed: false,
+    })
+      .then((res) => {
+        setTodos(prevTodos => [...prevTodos, res]);
+        setIsDisabled(false);
+        setTempTodo(null);
+      })
+      .catch(() => {
+        setErrorMessage(ErrorTypes.ADD);
+        showAndDeleteError();
+        setIsDisabled(false);
+        setTempTodo(null);
```

Comment:
`setIsDisabled(false);
   setTempTodo(null);
`
is duplicated in then and catch, so make sense to move it to the finally